graphs/models.py

```diff
@@ -310,6 +310,16 @@ class Edge(Base):
             ForeignKeyConstraint([user_id, graph_id, tail_node_id], [Node.user_id, Node.graph_id, Node.node_id], ondelete="CASCADE", onupdate="CASCADE"), {})
     #no relationship specified
 
+class share_graph_event(Base):
+    __tablename__ = 'share_graph_event'
+    id = Column(Integer, autoincrement=True, primary_key=True)
+    graph_id = Column(String, ForeignKey('graph.graph_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    owner_id = Column(String, ForeignKey('group.owner_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    group_id = Column(String, ForeignKey('group.group_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    member_id = Column(String, ForeignKey('user.user_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    share_time = Column(TIMESTAMP, nullable = False)
+    is_active = Column(Boolean)
```

Comment:
Don't you need a unique constraint on (graph_id, owner_id, group_id, member_id)? In other words, an owner can  share a graph with a group-member pair only once? 

Are you not adding the unique constraint because you are thinking of someone unsharing a graph and then sharing it again?